llvm/lib/Target/RISCV/RISCVOptWInstrs.cpp

```diff
@@ -711,7 +774,12 @@ bool RISCVOptWInstrs::runOnMachineFunction(MachineFunction &MF) {
 
   bool MadeChange = false;
   MadeChange |= removeSExtWInstrs(MF, TII, ST, MRI);
-  MadeChange |= stripWSuffixes(MF, TII, ST, MRI);
+
+  if (!(DisableStripWSuffix || ST.preferWInst()))
```

Comment:
These flags should not override the target preferences. DisableStripWSuffix was intended to disable the optimization if we found a bug.